client/src/components/Nav/Settings.jsx

```diff
@@ -13,14 +12,15 @@ export default function Settings({ open, onOpenChange }) {
   const [isMobile, setIsMobile] = useState(false);
   const lang = useRecoilValue(store.lang);
 
-  // check if mobile dynamically and update
+  // Check if mobile dynamically and update
   useEffect(() => {
     const checkMobile = () => {
       if (window.innerWidth <= 768) {
         setIsMobile(true);
       } else {
         setIsMobile(false);
       }
+      setIsMobile(window.innerWidth <= 768);
```

Comment:
now is a good opportunity to use my new `useMediaQuery` hook instead of the useEffect here

```js
import { useMediaQuery } from '~/hooks';
const isMobile = useMediaQuery('(max-width: 768px)'); // this is eq. to "md" breakpoint in tailwind
```

no more useEffect needed here, but if you still need a side effect, it would be like in` client\src\components\Input\ModelSelect\Plugins.tsx`

```js
  useEffect(() => {
    if (isSmallScreen) {
      setVisibility(false);
    }
  }, [isSmallScreen]);
```

Reply:
I just realized there's a bug here so you should use my new hook and get rid of the original useEffect